docs/pages/versions/unversioned/sdk/device.md

```diff
@@ -0,0 +1,273 @@
+- `Device.getUserAgentAsync()`
+- `Device.getTotalDiskCapacityAsync()`
+- `Device.getSerialNumberAsync()` (Android only)
+
+## Methods
+
+### `Device.hasNotch()`
+
+Tells if the device has a notch.
+
+#### Returns
+
+A boolean that represents the support for notch display.
+
+**Examples**
+
+```js
+import { hasNotch } from 'expo-device';
+
+hasNotch(); // true or false
+```
+
+### `Device.getUserAgentAsync()`
+
+Gets the device User Agent.
+
+#### Returns
+
+A Promise that resolves to a string of User Agent.
+
+**Examples**
+
+```js
+Device.getUserAgentAsync().then(userAgent => {
+  //Dalvik/2.1.0 (Linux; U; Android 9; Pixel 2 Build/PQ3A.190505.001)
+});
+```
+
+### `Device.getFreeDiskStorageAsync()`
+
+Gets available storage size, in bytes.
+
+#### Returns
+
+A promise of number that represents the storage size.
+
+**Examples**
+
+```js
+Device.getFreeDiskStorageAsync().then(storage => {
+  //5608296448
+});
+```
+
+### `Device.getTotalDiskCapacityAsync()`
+
+Gets full disk storage size, in bytes.
+
+#### Returns
+
+A promise of number that represents the full disk storage size.
+
+**Examples**
+
+```js
+Device.getTotalDiskCapacityAsync().then(storage => {
+  //17179869184
+});
+```
+
+### `Device.getIpAddressAsync()`
+
+Gets the device current IP address.
+
+#### Returns
+
+A Promise that resolves to a string of IP address.
+
+**Examples**
+
+```js
+Device.getIpAddressAsync().then(ip => {
+  // "92.168.32.44"
+});
+```
+
+### `Device.getMACAddressAsync(interfaceName?: string)`
+
+Gets the network adapter MAC address.
+
+#### Arguments (Android Only)
+
+- **interfaceName (_string_)** -- A string representing interface name (`eth0`, `wlan0`) or `null`, meaning the method should fetch MAC address of the first available interface. (On iOS this argument is ignored.)
+
+#### Returns
+
+A Promise that resolves to a string of the network adapter MAC address or empty string if there's no such address matching the interface.
+
+**Examples**
+
+```js
+//iOS
+Device.getMACAddressAsync().then(mac => {
+  // "E5:12:D8:E5:69:97"
+});
+
+//Android
+Device.getMACAddressAsync('wlan0').then(mac => {
+  // "E5:12:D8:E5:69:97"
+});
+```
+
+### `Device.getSerialNumberAsync()` (Android only)
+
+Gets the device serial number.
+
+Would prompt the user for permission to read the phone to access the phone number.
+
+#### Returns
+
+A Promise that resolves to a string of serial number of current device.
+
+**Examples**
+
+```js
+Device.getSerialNumberAsync().then(serialNumber => {
+  // FA81H1A01245
+});
+```
+
+### `Device.isAirplaneModeEnabledAsync()` (Android Only)
+
+Tells if the device is in airplane mode.
+
+#### Returns
+
+Returns a Promise that resolves to the `boolean` value for whether the device is in airplane mode or not.
+
+**Examples**
+
+```js
+Device.isAirplaneModeEnabledAsync().then(airPlaneModeOn => {
+  // false
+});
+```
+
+### `Device.hasSystemFeatureAsync(feature)` (Android Only)
+
+Tells if the device has a specific system feature.
```

Comment:
What does this mean? It's not clear what "system feature" means and how you would figure out what argument values to pass in.

Reply:
I added another function called `getSystemAvailableFeaturesAsync` to return all available system features on the device as a string[]. Will add it to the docs as well.